### PR TITLE
[Foundation] adjust inline of append and initialization functions

### PR DESCRIPTION
**Pull request by phausler**
Use _copyContents(initialzing:) to avoid performance regressions while using generic collection/sequence APIs

### CODE REVIEW

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by phausler** [comment]
@swift-ci please benchmark

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by phausler** [comment]
@swift-ci please benchmark

**Comment by phausler** [comment]
@swift-ci please benchmark

File: stdlib/public/SDK/Foundation/Data.swift [hunk 1]
```diff
@@ -1897,13 +1867,7 @@ extension Data {
 
 /// Provides bridging functionality for struct Data to class NSData and vice-versa.
 
-#if DEPLOYMENT_RUNTIME_SWIFT
```
Comment (parkera) [inline stdlib/public/SDK/Foundation/Data.swift]: Is this change orthogonal to the copyContents one?

File: stdlib/public/SDK/Foundation/Data.swift [hunk 2]
```diff
@@ -1260,30 +1238,20 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
...
-            self.init(backing: backing, range: 0..<len)
-        } else {
-            let underestimatedCount = elements.underestimatedCount
-            self.init(count: underestimatedCount)
-            
-            let (endIterator, _) = UnsafeMutableBufferPointer(start: _backing._bytes?.assumingMemoryBound(to: UInt8.self), count: underestimatedCount).initialize(from: elements)
-            var iter = endIterator
-            while let byte = iter.next() { self.append(byte) }
+        let backing = _DataStorage(capacity: Swift.max(elements.underestimatedCount, 1))
+        var (iter, endIndex) = elements._copyContents(initializing: UnsafeMutableBufferPointer(start: backing._bytes?.bindMemory(to: UInt8.self, capacity: backing._capacity), count: backing._capacity))
```
Comment (parkera) [inline stdlib/public/SDK/Foundation/Data.swift]: cc @atrick to double check the bind here

File: stdlib/public/SDK/Foundation/Data.swift [hunk 3]
```diff
@@ -1260,30 +1238,20 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     }
     
     // slightly faster paths for common sequences
-    
+    @inlinable
```
Comment (parkera) [inline stdlib/public/SDK/Foundation/Data.swift]: Are you sure this `@inlinable` doesn't expose the `_DataStorage` type?

File: stdlib/public/SDK/Foundation/Data.swift [hunk 4]
```diff
@@ -1122,28 +1122,6 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         _sliceRange = 0..<count
     }
     
-    /// Initialize a `Data` with the contents of an Array.
-    ///
-    /// - parameter bytes: An array of bytes to copy.
-    public init(bytes: Array<UInt8>) {
```
Comment (parkera) [inline stdlib/public/SDK/Foundation/Data.swift]: The Data perf tests have some cases that cover these specific types, right?

File: stdlib/public/SDK/Foundation/Data.swift (hunk 3 above)
Comment (phausler) [reply stdlib/public/SDK/Foundation/Data.swift]: @airspeedswift was telling me this is ok (similar to how Array does it)

File: stdlib/public/SDK/Foundation/Data.swift (hunk 1 above)
Comment (phausler) [reply stdlib/public/SDK/Foundation/Data.swift]: Correct but it allows me to iterate on sclf for validating this easily

File: stdlib/public/SDK/Foundation/Data.swift (hunk 4 above)
Comment (phausler) [reply stdlib/public/SDK/Foundation/Data.swift]: Yes it does

**Comment by phausler** [comment]
@swift-ci please benchmark

**Comment by phausler** [comment]
@swift-ci please smoke test

**Comment by phausler** [comment]
@swift-ci please benchmark

**Pull request merged**